docs/aws/audit/kubernetesmonitoring/rules/ecs_cluster_container_insights_enable_remediation.mdx

```diff
@@ -1,6 +1,231 @@
+   - In the cluster details page, choose the **Update Cluster** button.
+   - In the **Monitoring** section, check the box for **Enable Container Insights**.
+
+3. **Review and Update:**
+   - Review the settings to ensure Container Insights is enabled.
+   - Choose **Update** to apply the changes.
+
+4. **Verify Configuration:**
+   - After updating, go to the **CloudWatch** service in the AWS Management Console.
+   - Navigate to **Container Insights** under the **Metrics** section to verify that metrics are being collected for your ECS cluster.
+
+By following these steps, you can ensure that ECS Container Insights are enabled, which helps in monitoring and troubleshooting your Kubernetes workloads on AWS.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the misconfiguration of ECS Container Insights not being enabled in Kubernetes using AWS CLI, follow these steps:
+
+1. **Update the AWS CLI to the Latest Version:**
+   Ensure you have the latest version of the AWS CLI installed to access the most recent features and commands.
+   ```sh
+   aws --version
+   ```
+
+2. **Enable Container Insights for ECS Clusters:**
+   Use the `update-cluster-settings` command to enable Container Insights for your ECS cluster.
+   ```sh
+   aws ecs update-cluster-settings --cluster <your-cluster-name> --settings name=containerInsights,value=enabled
+   ```
+
+3. **Verify the Cluster Settings:**
+   Confirm that Container Insights has been enabled by describing the cluster settings.
+   ```sh
+   aws ecs describe-clusters --clusters <your-cluster-name> --query "clusters[0].settings"
+   ```
+
+4. **Configure CloudWatch Agent for Kubernetes:**
+   Ensure the CloudWatch agent is configured to collect metrics from your Kubernetes cluster. This involves creating a CloudWatch agent configuration file and applying it to your cluster.
+   ```sh
+   kubectl apply -f https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/master/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/quick-start/cwagent/cwagent-daemonset.yaml
+   ```
+
+By following these steps, you can ensure that ECS Container Insights are enabled for your Kubernetes clusters using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of ECS Container Insights not being enabled in Kubernetes using Python scripts, you can follow these steps:
+
+### 1. **Set Up AWS SDK for Python (Boto3)**
+First, ensure you have the AWS SDK for Python (Boto3) installed. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### 2. **Create a Boto3 Session**
+Create a Boto3 session to interact with AWS services. This will allow you to authenticate and make API calls.
+
+```python
+import boto3
+
+# Create a session using your AWS credentials
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>